### ISSUE TITLE
[NFR] Make SplitRuleCollection Countable

### ISSUE THREAD

**Issue by williamokano** [pagarme/pagarme-php#133]
IMHO the `SplitRuleCollection` could implement the `Countable`. Today I need to calculate how many `Payables` a transaction created, so I had to get the `Transaction` and then iterate over the `SplitRules` to get how many splits the transaction have.

```php
$transaction = $pagarme->transaction->get($transactionId);
$splitRulesQtt = 0;
foreach ($transaction->getSplitRules() as $splitRule) {
    $splitRulesQtt++;
}

$payablesQtt = $transaction->getInstallments() * $splitRulesQtt;
```
**Proposal**
```php
class SplitRuleCollection implements \ArrayAccess, \Iterator, \Countable
{
    // ... Existing code

    /**
     * Count the number of items in the collection.
     *
     * @return int
     */
    public function count()
    {
        return count($this->rules);
    }
}
```

**Comment by devdrops** [comment]
@williamokano what are the advantages when implementing the `ArrayAccess` interface?

**Comment by williamokano** [comment]
@devdrops It already implements, I just added the `Countable`.

**Comment by devdrops** [comment]
Ouch, my bad! :laughing:

I'll check this issue later today. Could you please provide a pull request? That would be awesome!

**Comment by williamokano** [comment]
Sure

**Comment by devdrops** [comment]
Merged at [a708bce](https://github.com/pagarme/pagarme-php/commit/a708bce301c9c0985dd61e954b522171fe553642).
